src/transaction/log_manager.c

```diff
@@ -12950,33 +12950,51 @@ cdc_put_value_to_loginfo (db_value * new_value, char **data_ptr)
     case DB_TYPE_BIT:
     case DB_TYPE_VARBIT:
       {
-	char result[1024] = "\0";
+	char temp[1024];
+	char *result;
 	int length, n, count;
 	char *bitstring = NULL;
 	func_type = 7;
 
 	length = ((db_get_string_length (new_value) + 3) / 4) + 4;
 
-	bitstring = (char *) malloc (length);
-	if (bitstring == NULL)
+	if (length <= 1024)
```

Comment:
Need to think X' and ' and \0.

Reply:
Does +4 mean this consideration?

Reply:
yes, 
(db_get_string_length (new_value) + 3) / 4) means how many characters in bit/varbit
and + 4 means the count of "X' '\0"